File: SDK-64bit/doc/_downloads/almotion_changeposition.py
import sys
import time
from naoqi import ALProxy

if (len(sys.argv) < 2):
    print "Usage: 'python motion_changeposition.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALMotion", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALMotion"
    print "Error was: ",e
    sys.exit(1)
    
proxy.setStiffnesses("LArm", 1.0)

# Example showing how to move forward (2cm) "LArm". 
effectorName     = "LArm" 
space            = 2 # SPACE_NAO 
positionChange   = [0.05, 0.0, 0.0, 0.0, 0.0, 0.0] 
fractionMaxSpeed = 0.5 
axisMask         = 7 
proxy.changePosition(effectorName, space, positionChange, fractionMaxSpeed, axisMask)

time.sleep(1.0)

proxy.setStiffnesses("LArm", 0.0)

File: SDK-64bit/doc/_downloads/almotion_changeposition.cpp
#include <iostream>
#include <alproxies/almotionproxy.h>
#include <qi/os.hpp>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: motion_changeposition pIp"
              << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  AL::ALMotionProxy motion(pIp);

  // Example showing how to move forward (2cm) "LArm".
  std::string effectorName  = "LArm";
  int space                 = 2; // SPACE_NAO
  std::vector<float> positionChange(6, 0.0f);
  positionChange[0] = 0.02f;
  float fractionMaxSpeed    = 0.5f;
  int axisMask              = 7;
  // Setting effector stiffness on.
  motion.setStiffnesses(effectorName, 1.0f);
  motion.changePosition(effectorName, space, positionChange, fractionMaxSpeed, axisMask);

  qi::os::sleep(1.0f);

  // Setting left arm stiffness on.
  motion.setStiffnesses(effectorName, 0.0f);

  return 0;
}
